### PR TITLE
feat(CO-39): DB service for existing data

### PR DESCRIPTION
**Pull request by rrtoth**

### CODE REVIEW

**Review by lakospal0 — APPROVED** [review]

**Review by lakospal0 — DISMISSED** [review]

File: src/services/registration.js [hunk 1]
```diff
@@ -1,40 +1,65 @@
...
+const conn = mysql.createConnection({
+  host: process.env.DB_HOST,
+  user: process.env.DB_USER,
+  password: process.env.DB_PASS,
+  database: process.env.DB_NAME,
+  port: process.env.DB_PORT,
+  multipleStatements: true,
+});
+
+const checkUserName = (input) => {
```
Comment (matyaspeterfi) [inline src/services/registration.js]: Can you please somehow specify in the function name what it's checking the username for?

File: src/services/registration.js [hunk 2]
```diff
@@ -1,40 +1,65 @@
...
+  const user = /^[A-Za-z0-9]\w{0,}$/;
+
+  if (input && user.test(input)) {
+    return true;
   }
+  return false;
+};
+
+const checkPassword = (input) => {
```
Comment (matyaspeterfi) [inline src/services/registration.js]: Same as above, not sure what the function is checking the input for

**Review by gbr88 — CHANGES_REQUESTED** [review]
const conn = mysql.createConnection .... can be in separated file and just import it in those files where it needed.

File: src/services/registration.js (hunk 1 above)
Comment (rrtoth) [reply src/services/registration.js]: not my task but updated as it took like 2 seconds

File: src/services/registration.js (hunk 2 above)
Comment (rrtoth) [reply src/services/registration.js]: same

**Review by matyaspeterfi — DISMISSED** [review]

**Review by gbr88 — DISMISSED** [review]

**Review by matyaspeterfi — APPROVED** [review]

**Review by matyaspeterfi — CHANGES_REQUESTED** [review]
Lint fail on circleCi :(

**Review by matyaspeterfi — DISMISSED** [review]

File: src/controllers/loginController.js [hunk 3]
```diff
@@ -1,31 +1,20 @@
...
-    const notEnteredDataError = {
-      message: 'Please enter a Username and a Password!',
-    };
-    res.status(400).send(notEnteredDataError);
+  login(req, res) {
+    if (req.body.username && req.body.password) {
+      this.loginService.authorizeUser({
+        username: req.body.username,
+        password: req.body.password,
+      }).then(data => res.status(200).json({ userId: data }))
```
Comment (juhaszPeterDev) [inline src/controllers/loginController.js]: the answer object should not be created in the controller unless you want to attach metadata to it. if you do this then you wont be able to change the data structure so easily.

File: src/routes/index.js [hunk 4]
```diff
@@ -1,20 +1,38 @@
...
+
+if (process.env.NODE_ENV === 'test') {
+  useddb = require('../services/mockDb'); // eslint-disable-line
+}
+
+const heroService = new HeroService(useddb);
+const heroController = new HeroController(heroService);
+const registrationService = new RegistrationService(useddb);
+const registrationController = new RegistrationController(registrationService);
+const loginService = new LoginService(useddb);
```
Comment (juhaszPeterDev) [inline src/routes/index.js]: see later comment for this line

File: src/services/loginService.js [hunk 5]
```diff
@@ -0,0 +1,26 @@
+const RegistrationService = require('./registrationService');
+
+class LoginService {
+  constructor(conn) {
+    this.conn = conn;
+  }
+
+  async authorizeUser(input) {
+    const registrationService = new RegistrationService(this.conn);
```
Comment (juhaszPeterDev) [inline src/services/loginService.js]: this should be passed through constructor parameter as a dependency injection. you shouldn't create a new service instance every time even if it's not a problem in this case. this comment is referred from routes/index.js

**Review by juhaszPeterDev — CHANGES_REQUESTED** [review]

File: src/controllers/loginController.js (hunk 3 above)
Comment (rrtoth) [reply src/controllers/loginController.js]: corrected

File: src/routes/index.js (hunk 4 above)
Comment (rrtoth) [reply src/routes/index.js]: corrected

File: src/services/loginService.js (hunk 5 above)
Comment (rrtoth) [reply src/services/loginService.js]: corrected

**Review by matyaspeterfi — DISMISSED** [review]

File: src/controllers/loginController.js (hunk 3 above)
Comment (juhaszPeterDev) [reply src/controllers/loginController.js]: wrong line, wrong correction, the service call was fine, i commented the 12th line

**Review by matyaspeterfi — APPROVED** [review]

**Review by juhaszPeterDev — APPROVED** [review]

**Pull request merged**